### ISSUE TITLE
Jaki jest "Dodatkowy limit liczby zapytań dla użytkownika"?

### ISSUE THREAD

**Issue by jakubmaguza** [allegro/allegro-api#3345]
Temat dotyczy WebApi
Hej,
Wykorzystuję dwie metody z webapi - doGetSiteJournal i doGetItemsInfo.
Mianowicie to, co wypluje metoda nr.1 wrzucam to metody nr.2.
Przy małym ruchu na allegro, raczej nie ma problemu, ale w niedzielę wieczorem, kiedy dużo ludzi wrzuca nowe przedmioty, pojawia się bardzo dużo zapytań do serwera.
Aplikacja wysyła wtedy coraz więcej zapytań jednocześnie i losowo, co jakiś czas dostaje timeout.
Timeout ustawiłem po stronie klienta na 10 sekund.

Wydaje mi się, że problem pojawia się przez to:
https://developer.allegro.pl/about/#limits

Więc kilka pytań:
1. Jaki jest ten "Dodatkowy limit liczby zapytań dla użytkownika"?
2. Czy jest to limit czasowy/jednoczesnych zapytań?
3. Czy jest to per zalogowany user/clientID/IP?

**Comment by TimothyKoval** [comment]
[https://allegro.pl/webapi/faq.php#faq_1](https://allegro.pl/webapi/faq.php#faq_1)

> W WebAPI obowiązuje główny limit nakładany na klucz WebAPI - 9000 zapytań na minutę.
> 
> Jeśli go przekroczysz zablokujemy klucz WebAPI na minutę, a w odpowiedzi otrzymasz błąd: ERR_MINUTE_LIMIT_EXCEEDED.

**Comment by TimothyKoval** [comment]
Link który podałeś dotyczy REST API, nie WEB API

**Comment by jakubmaguza** [comment]
Tak, ale nie pytam o ogólny limit, tylko o szczegóły konkretnego zwalniania odpowiedzi po przekroczeniu liczby zapytań, o którym to limicie mówi tylko strona REST, ale zapewne dotyczy też WEB.

**Comment by TimothyKoval** [comment]
Na REST API allegro stosuje leaky bucket, nic mi o tym nie wiadomo w przypadku WEB API. Osobiście wywołuję średnio po 200 - 300 requestów na sekundę pomiędzy 3 serwerami (jeden klucz) i nie zauważam żadnego lejka poza tym na moich dyskach :)

**Comment by PrzemyslawLukanowski** [comment]
Leaky Bucket dotyczy tylko REST API. Jedyny limit dla WebAPI to ten, o którym mowa w linku udostępnionym wyżej - 9000 zapytań na minutę.

**Comment by jakubmaguza** [comment]
1. Okej, w takim razie co z zapytaniami, dla których dostaję timeout? Możecie sprawdzić co się dzieje? Czy te 10 sekund to zbyt krótko i niektóre odpowiedzi muszą przychodzić później?
2. Czasem dostaję też bardzo dziwną odpowiedź, że zapytanie zostało wysłane po HTTP. To jest niemożliwe, bo wszystkie zapytania są tworzone w ten sam sposób, więc albo wszystkie szły by po HTTP, albo wszystkie po HTTPS. Czy możecie wyśledzić te zapytania?

**Comment by PrzemyslawLukanowski** [comment]
Prześlij proszę przykładowe logi przez nasz [formularz kontaktowy ](https://allegro.pl/pomoc/kontakt?kategoria=0cad5ed6-ca26-4669-8ff0-6a7da503fe80&subjectId=8b7d668e-6f52-47e4-8c6b-cfb5e985789a), w zgłoszeniu dopisz GitHub #3345. Zweryfikuję ten temat i wrócę z odpowiedzią.

**Comment by jakubmaguza** [comment]
@PrzemyslawLukanowski Wysłałem przed chwilą zgłoszenie :)

**Comment by jakubmaguza** [comment]
Błąd był gdzieś w architekturze programu/systemu, bo standardowo - na jednym komputerze działało, na drugim nie.

W każdym razie, niedługo wszystko przejdzie na REST API, czy w takim wypadku możecie powiedzieć jaki jest ten dodatkowy limit, po którym będziecie opóźniać czas odpowiedzi? I jak bardzo będziecie go opóźniać/na jak długo?

**Comment by PrzemyslawLukanowski** [comment]
Dodatkowy limit dotyczy tylko wybranych endpointów, np. [GET /sale/products](https://developer.allegro.pl/documentation/#operation/getSaleProducts) . Polega na tym, że po przekroczeniu dozwolonej liczby RPM wydłużymy czas odpowiedzi. Ze względów bezpieczeństwa nie możemy udostępnić informacji, jak długie są to opóźnienia. W takiej sytuacji należy zmniejszyć liczbę requestów.

**Comment by jakubmaguza** [comment]
Okej, dzięki za informację.
